build.assets/docker/os-rootfs/etc/node-problem-detector/kernel-monitor.json

```diff
@@ -0,0 +1,64 @@
+{
+  "plugin": "kmsg",
+  "logPath": "/dev/kmsg",
+  "lookback": "5m",
+  "bufferSize": 10,
+  "source": "kernel-monitor",
+  "conditions": [
+    {
+      "type": "KernelDeadlock",
+      "reason": "KernelHasNoDeadlock",
+      "message": "kernel has no deadlock"
+    },
+    {
+      "type": "ReadonlyFilesystem",
+      "reason": "FilesystemIsNotReadOnly",
+      "message": "filesystem is not read-only"
+    }
+  ],
+  "rules": [
+    {
+      "type": "temporary",
+      "reason": "OOMKilling",
+      "pattern": "Kill process \\d+ (.+) score \\d+ or sacrifice child\\nKilled process \\d+ (.+) total-vm:\\d+kB, anon-rss:\\d+kB, file-rss:\\d+kB.*"
+    },
+    {
+      "type": "temporary",
+      "reason": "TaskHung",
+      "pattern": "task \\S+:\\w+ blocked for more than \\w+ seconds\\."
```

Comment:
nit: the dot at the end is probably not necessary in the pattern

Reply:
I think it is necessary. IIUC how NPD's matching works, it has to match the entire strings, beginning-to-end. Again, this is the default configuration from their repo.